lib/ecto/adapters/sql.ex

```diff
@@ -124,6 +126,24 @@ defmodule Ecto.Adapters.SQL do
         :ok
       end
 
+      def do_get_sources(meta) do
```

Comment:
We don't need to get all sources, only the first one (which is the one used in `from`).